cpp/src/io/parquet/page_enc.cu

```diff
@@ -586,10 +584,14 @@ inline __device__ void PackLiteralsRoundRobin(
 {
   // Scratch space to temporarily write to. Needed because we will use atomics to write 32 bit
   // words but the destination mem may not be a multiple of 4 bytes.
-  // TODO (dm): This assumes blockdim = 128 and max bits per value = 16. Reduce magic numbers.
-  // To allow up to 24 bit this needs to be sized at 96 words.
-  __shared__ uint32_t scratch[64];
-  if (t < 64) { scratch[t] = 0; }
+  // TODO (dm): This assumes blockdim = 128. Reduce magic numbers.
+  constexpr uint32_t NUM_THREADS  = 128;  // this needs to match gpuEncodePages block_size parameter
+  constexpr uint32_t NUM_BYTES    = (NUM_THREADS * MAX_DICT_BITS) >> 3;
+  constexpr uint32_t SCRATCH_SIZE = NUM_BYTES / sizeof(uint32_t);
+  __shared__ uint32_t scratch[SCRATCH_SIZE];
+  for (uint32_t i = 0; i < SCRATCH_SIZE; i += NUM_THREADS) {
+    if ((i + t) < SCRATCH_SIZE) { scratch[i + t] = 0; }
+  }
```

Comment:
Just curious, would this be any better/worse?
```
  for (uint32_t i = t; i < SCRATCH_SIZE; i += NUM_THREADS) {
    scratch[i] = 0;
  }
```

Reply:
Yeah, that's probably better.  No conditional needed.